dsd/constraints.py

```diff
@@ -636,22 +639,30 @@ def _get_next_sequence_satisfying_numpy_constraints(self, rng: np.random.Generat
         return sequence
 
     def _generate_sequences_satisfying_numpy_constraints(self, rng: np.random.Generator) -> List[str]:
+        global updated_num_gen_sequences # 10 ** 5 initial value
         bases = self._bases_to_use()
         length = self.length
         use_random_subset = length > _length_threshold_numpy
         if not use_random_subset:
             seqs = dn.DNASeqList(length=length, alphabet=bases, shuffle=True, rng=rng)
             num_starting_seqs = seqs.numseqs
         else:
-            num_starting_seqs = num_random_sequences_to_generate_at_once
+            num_starting_seqs = updated_num_gen_sequences
             seqs = dn.DNASeqList(length=length, alphabet=bases, shuffle=True,
                                  num_random_seqs=num_starting_seqs, rng=rng)
-
         seqs_satisfying_numpy_constraints = self._filter_numpy_constraints(seqs)
-
-        if seqs_satisfying_numpy_constraints.numseqs == 0:
-            raise ValueError('no sequences passed the numpy constraints')
-
+        while seqs_satisfying_numpy_constraints.numseqs == 0 and updated_num_gen_sequences < 4 ** 15:
```

Comment:
Has this been tested?

I'd try testing it with smaller values of `num_random_sequences_to_generate_at_once`, with fairly strict constraints, so that it actually has to run through at least one iteration of the loop until `updated_num_gen_sequences` gets big enough.